larray/core/array.py

```diff
@@ -2219,11 +2219,22 @@ def __setitem__(self, key, value, collapse_slices=True):
             else:
                 axes = self._get_axes_from_translated_key(translated_key)
             value = value.broadcast_with(axes)
+            # replace incomprehensible error message
+            # "could not broadcast input array from shape XX into shape YY"
+            # for users by "incompatible axes"
+            try:
+                data[cross_key] = value
+            except ValueError as err:
+                new_message = "Incompatible axes. Values from input array with axes\n" \
+                              "{!r}\ncannot be set to subset with axes\n{!r}".format(value.axes, self[key].axes)
+                if PY2:
+                    err.message = new_message
+                    raise
+                else:
+                    raise ValueError(new_message) from err
```

Comment:
The problem is only present if there are more axes on the value than the target, so I would have written something like:

```python
extra_axes = value.axes - axes
if extra_axes:
    text = 'axes are' if len(extra_axes) > 1 else 'axis is'
    raise ValueError("Value {} {} not present in target subset {}. A value can only have the same axes or fewer axes than the subset being targeted".format(extra_axes, text, axes))
```
which would yield, for example:

> Value {c} axis is not present in target subset {a, b}. A value can only have the same axes or fewer axes than the subset being targeted